cartographer/mapping/internal/constraints/constraint_builder_2d.cc

```diff
@@ -258,19 +258,33 @@ void ConstraintBuilder2D::ComputeConstraint(
   // Use the CSM estimate as both the initial and previous pose. This has the
   // effect that, in the absence of better information, we prefer the original
   // CSM estimate.
+
   ceres::Solver::Summary unused_summary;
+  Eigen::Matrix3d seedless_precision;
```

Comment:
May consider initializing to zeros here. Ceres also uses sparse matrices(sometimes) and depending on the implementation, there may be a condition in which the `Evaluate` may only assign to the non-zero entries of the matrix leaving garbage value for the zero entries.

Reply:
Hm. Good catch. Will do.